crates/sui-graphql-rpc/src/types/move_package.rs

```diff
@@ -187,7 +187,13 @@ impl MovePackage {
     fn parsed_package(&self) -> Result<ParsedMovePackage, Error> {
         // TODO: Leverage the package cache (attempt to read from it, and if that doesn't succeed,
         // write back the parsed Package to the cache as well.)
-        ParsedMovePackage::read(&self.super_.native)
+        let Some(native) = self.super_.native_impl() else {
+            return Err(Error::Internal(
+                "This should be unreachable as we cannot downcast if native is None".to_string(),
```

Comment:
It would be clearer if this message explained what went wrong instead of why it shouldn't have happened.

```suggestion
                "No native representation of package to parse.".to_string(),
```